### DISCUSSION TITLE
Ruby error

### DISCUSSION THREAD

**Discussion by Modrx** [mastodon/mastodon#25135] [Q&A]
### Steps to reproduce the problem

When I run the following command:

RAILS_ENV=production bundle exec rake mastodon:setup

### Expected behaviour

no errors

### Actual behaviour

showing warning

### Detailed description

I am getting the following errors:

/usr/lib/ruby/2.7.0/net/protocol.rb:66: warning: already initialized constant Net::ProtocRetryError
/live/vendor/bundle/ruby/2.7.0/gems/net-protocol-0.2.1/lib/net/protocol.rb:68: warning: previous definition of ProtocRetryError was here
/usr/lib/ruby/2.7.0/net/protocol.rb:206: warning: already initialized constant Net::BufferedIO::BUFSIZE
/live/vendor/bundle/ruby/2.7.0/gems/net-protocol-0.2.1/lib/net/protocol.rb:214: warning: previous definition of BUFSIZE was here
/usr/lib/ruby/2.7.0/net/protocol.rb:503: warning: already initialized constant Net::NetPrivate::Socket
/live/vendor/bundle/ruby/2.7.0/gems/net-protocol-0.2.1/lib/net/protocol.rb:541: warning: previous definition of Socket was here

### Specifications

Mastodon v4.1.2
Ruby 3.2.2
NodeJS 19.9.0
Debian 11

**Comment by renchap** [comment]
You say you are running Ruby 3.2.2, but your gems seem to come from a Ruby 2.7 install:

`/usr/lib/ruby/2.7.0/net/protocol.rb:66: warning: already initialized constant Net::ProtocRetryError`

Your Ruby setup is probably wrong and is causing this.

**Comment by Modrx** [comment]
I have both versions installed, issue is resolved, thank you!
